java/code/src/com/redhat/rhn/frontend/xmlrpc/configchannel/ConfigChannelHandler.java

```diff
@@ -389,6 +389,7 @@ public ConfigChannel update(User loggedInUser, String label,
      *  #struct_begin("path info")
      *      #prop_desc("string","contents", "Contents of the init.sls file")
      *      #prop_desc("boolean","contents_enc64", "Identifies base64 encoded content(default: disabled)")
+     *      #prop_desc("int", "revision", "next revision number, auto increment for null")
```

Comment:
What are the pros and cons to add this as a field in this map rather than a full new parameter?

Reply:
**PROS**
- Consistency with the syntax already used on `configchannel.createOrUpdatePath`, which is the equivalent method for updating "files" (instead of "states")
- No need to overload this method to guarantee that we are not breaking the current method signature
- Do not pollute method signature with optional parameters
- Since `contents` is a required field, and lives inside the map, It doesn't seem right to keep the **required** `contents` field in the map and the **optional** `revision` field as a new parameter (since `revision` is less important and less used)

**CONS**
- Field is not explicitly declared, so we have to rely on documentation to check which "fields" are accepted, but unfortunately Java does not have a great way to deal with optional parameters (requires method overload for that)

Let me know if you prefer a new parameter, so I can update the PR accordingly.

Reply:
> **PROS**
> 
>     * Consistency with the syntax already used on `configchannel.createOrUpdatePath`, which is the equivalent method for updating "files" (instead of "states")

Good point.

>     * No need to overload this method to guarantee that we are not breaking the current method signature

Overloading is not a problem with Java and XMLRPC, so that is not a concern I would say.

>     * Do not pollute method signature with optional parameters
> 
>     * Since `contents` is a required field, and lives inside the map, It doesn't seem right to keep the **required** `contents` field in the map and the **optional** `revision` field as a new parameter (since `revision` is less important and less used)
> 
> 
> **CONS**
> 
>     * Field is not explicitly declared, so we have to rely on documentation to check which "fields" are accepted, but unfortunately Java does not have a great way to deal with optional parameters (requires method overload for that)
> 
> 
> Let me know if you prefer a new parameter, so I can update the PR accordingly.

I am convinced about the pros, OK with the current direction :+1: